client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js

```diff
@@ -1,242 +1,255 @@
-        // and validation can always be done by clicking
-          self.order.autoPay = false;
-        },
-      );
-    }).catch((err) => {
-      TucToast.error([$translate.instant('telephony_alias_portability_order_error'), _.get(err, 'data.message')].join(' '));
-      return $q.reject(err);
-    }).finally(() => {
-      self.order.isOrdering = false;
+      self.onSDATypeChange = function () {
+        self.order.socialReason = self.isSDA ? 'corporation' : 'individual';
+      };
+
+      self.openDesireDatePicker = function (ev) {
+        ev.preventDefault();
+        ev.stopPropagation();
+        self.desireDatePickerOpened = true;
+      };
+
+      // select number corresponding country automatically
+      self.onNumberChange = function () {
+        const number = self.normalizeNumber(self.order.callNumber);
+        if (_.startsWith(number, '0033')) {
+          self.order.country = 'france';
+        } else if (_.startsWith(number, '0032')) {
+          self.order.country = 'belgium';
+          self.order.rio = null;
+        } else if (_.startsWith(number, '0041')) {
+          self.order.country = 'switzerland';
+          self.order.rio = null;
+        }
+
+        // handle special number
+        self.isSpecialNumber = _.some(
+          specialNumberPrefix[self.order.country],
+          prefix => _.startsWith(number, prefix),
+        );
+        self.order.specialNumberCategory = self.isSpecialNumber
+          ? _.first(self.typologies[self.order.country]).value : null;
+        self.order.type = self.isSpecialNumber ? 'special' : 'landline';
+
+        self.order.translatedCountry = $translate.instant(`telephony_alias_portability_order_contact_country_${self.order.country}`);
+      };
+
+      self.onChooseRedirectToLine = function (result) {
+        self.order.lineToRedirectAliasTo = result.serviceName;
+        self.order.lineToRedirectAliasToDescription = result.description;
+      };
+
+      // add sdaNumberToAdd number to numbersList
+      self.addSdaNumber = function () {
+        self.order.numbersList.push(self.order.sdaNumberToAdd);
+        self.order.numbersList = _.uniq(self.order.numbersList);
+        self.order.sdaNumberToAdd = null;
+      };
+
+      // remove given number from numbersList
+      self.removeSdaNumber = function (number) {
+        self.order.numbersList = _.pull(self.order.numbersList, number);
+      };
+
+      // normalize number : replace +33 by 0033
+      self.normalizeNumber = function (numberParam) {
+        let number = numberParam;
+        if (number) {
+          number = number.replace(/^\+/, '00');
+        }
+        return number;
+      };
+
+      self.goToConfigStep = function () {
+        self.order.addressTooLong = false;
+
+        if ((_.get(self.order, 'streetName', '').length + _.get(self.order, 'streetNumber', '').length + _.get(self.order, 'streetNumberExtra', '').length + _.get(self.order, 'streetType', '').length) >= 35) {
```

Comment:
You can do this this way if you want : 
```suggestion
        if (`${_.get(self.order, 'streetName', '')}${_.get(self.order, 'streetNumber', '')}${_.get(self.order, 'streetNumberExtra', '')}${_.get(self.order, 'streetType', '')}`.length >= 35) {
```

Reply:
previous code